extensions/standard-processors/processors/ListenSyslog.cpp

```diff
@@ -17,318 +14,283 @@
-        logger_->log_error("ListenSysLog Server socket bind failed");
-        break;
-      }
-      if (_protocol == "TCP")
-        listen(sockfd, 5);
-      _serverSocket = sockfd;
-      logger_->log_info("ListenSysLog Server socket %d bind OK to port %d", _serverSocket, portno);
-    }
-    FD_ZERO(&_readfds);
-    FD_SET(_serverSocket, &_readfds);
-    _maxFds = _serverSocket;
-    std::vector<int>::iterator it;
-    for (it = _clientSockets.begin(); it != _clientSockets.end(); ++it) {
-      int clientSocket = *it;
-      if (clientSocket >= _maxFds)
-        _maxFds = clientSocket;
-      FD_SET(clientSocket, &_readfds);
-    }
-    fd_set fds;
-    struct timeval tv;
-    int retval;
-    fds = _readfds;
-    tv.tv_sec = 0;
-    // 100 msec
-    tv.tv_usec = 100000;
-    retval = select(_maxFds + 1, &fds, NULL, NULL, &tv);
-    if (retval < 0)
-      break;
-    if (retval == 0)
-      continue;
-    if (FD_ISSET(_serverSocket, &fds)) {
-      // server socket, either we have UDP datagram or TCP connection request
-      if (_protocol == "TCP") {
-        socklen_t clilen;
-        struct sockaddr_in cli_addr;
-        clilen = sizeof(cli_addr);
-        int newsockfd = accept(_serverSocket, reinterpret_cast<struct sockaddr *>(&cli_addr), &clilen);
-        if (newsockfd > 0) {
-          if (_clientSockets.size() < (uint64_t) _maxConnections) {
-            _clientSockets.push_back(newsockfd);
-            logger_->log_info("ListenSysLog new client socket %d connection", newsockfd);
-            continue;
-          } else {
-            close(newsockfd);
-          }
-        }
-      } else {
-        socklen_t clilen;
-        struct sockaddr_in cli_addr;
-        clilen = sizeof(cli_addr);
-        int recvlen = recvfrom(_serverSocket, _buffer, sizeof(_buffer), 0, (struct sockaddr *) &cli_addr, &clilen);
-        if (recvlen > 0 && (uint64_t) (recvlen + getEventQueueByteSize()) <= static_cast<uint64_t>(_recvBufSize)) {
-          uint8_t *payload = new uint8_t[recvlen];
-          memcpy(payload, _buffer, recvlen);
-          putEvent(payload, recvlen);
-        }
-      }
-    }
-    it = _clientSockets.begin();
-    while (it != _clientSockets.end()) {
-      int clientSocket = *it;
-      if (FD_ISSET(clientSocket, &fds)) {
-        int recvlen = readline(clientSocket, _buffer, sizeof(_buffer));
-        if (recvlen <= 0) {
-          close(clientSocket);
-          logger_->log_debug("ListenSysLog client socket %d close", clientSocket);
-          it = _clientSockets.erase(it);
-        } else {
-          if ((uint64_t) (recvlen + getEventQueueByteSize()) <= static_cast<uint64_t>(_recvBufSize)) {
-            uint8_t *payload = new uint8_t[recvlen];
-            memcpy(payload, _buffer, recvlen);
-            putEvent(payload, recvlen);
-          }
-          ++it;
-        }
-      }
-    }
+  if (!context->getProperty(MaxBatchSize.getName(), max_batch_size_) || max_batch_size_ <= 0) {
```

Comment:
I think having set `withDefaultValue<uint64_t>` the `UNSIGNED_LONG_VALIDATOR` is used and will reject this branch

Reply:
:+1: I've changed it to a custom validator https://github.com/apache/nifi-minifi-cpp/pull/1294/commits/027fea1d65bd423940edd81d598cef68946e2202#diff-a22ae2c37869a184bd936ac719f98ae8bbdb1b8d2ad5b3e6fc48ea9207fe530bR43